sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetTest.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.shardingsphere.shardingjdbc.jdbc.core.resultset;
+
+import org.apache.shardingsphere.shardingjdbc.common.base.AbstractEncryptJDBCDatabaseAndTableTest;
+import org.junit.Test;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class EncryptResultSetTest extends AbstractEncryptJDBCDatabaseAndTableTest {
+
+    private static final String SELECT_SQL_TO_ASSERT = "SELECT id, cipher_pwd, plain_pwd FROM t_encrypt";
+
+    @Test
+    public void assertResultSetIsBeforeFirst() throws SQLException {
+        try (Statement statement = getEncryptConnection().createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE,
+                ResultSet.CONCUR_READ_ONLY)) {
+            ResultSet resultSet = statement.executeQuery(SELECT_SQL_TO_ASSERT);
+            assertTrue(resultSet.isBeforeFirst());
+            resultSet.next();
+            assertEquals(1, resultSet.getRow());
+            resultSet.afterLast();
+            assertTrue(resultSet.isAfterLast());
+            resultSet.beforeFirst();
+            assertTrue(resultSet.isBeforeFirst());
```

Comment:
This test method test more than one case, let's separate it to multiple methods.
Just follow SRP.